core/src/main/scala/com/avast/clients/storage/compression/ZstdDecompressOutputStream.scala

```diff
@@ -0,0 +1,52 @@
+package com.avast.clients.storage.compression
+
+import com.github.luben.zstd.{ZstdDecompressCtx, ZstdInputStream}
+
+import java.io.OutputStream
+import java.nio.ByteBuffer
+import java.nio.channels.Channels
+
+class ZstdDecompressOutputStream(outputStream: OutputStream) extends OutputStream {
+  private val decompressCtx = new ZstdDecompressCtx()
+  private val outputChannel = Channels.newChannel(outputStream)
+  private val closed = false
+
+  override def write(chunk: Array[Byte]): Unit = {
+    if (closed) {
+      throw new IllegalStateException("Stream is closed")
+    }
+
+    val inputBuffer = ByteBuffer.allocateDirect(chunk.length)
+    val outputBuffer = ByteBuffer.allocateDirect(ZstdInputStream.recommendedDOutSize().toInt)
+
+    inputBuffer.put(chunk)
+    inputBuffer.flip()
```

Comment:
I believe this does the same as `reset` as the capacity/limit of the ByteBuffer is already set to the right position. However, if you use `ByteBuffer.wrap` as suggested above, you will need to call neither `put` nor `flip`.

Reply:
`reset` doesn't work, `rewind` does, but it is almost the same as the `flip`.